serial/src/main/java/org/jboss/marshalling/serial/SerialUnmarshaller.java

```diff
@@ -342,6 +348,44 @@ Object doReadObject(int leadByte, final boolean unshared) throws IOException, Cl
         }
     }
 
+    private Object doReadRecord(final SerializableClass sc) throws ClassNotFoundException, IOException {
+        final Object[] values = new Object[sc.getFields().length];
+        for (SerializableField serializableField : sc.getFields()) {
+            switch (serializableField.getKind()) {
+                case BOOLEAN:
+                    values[serializableField.getRecordComponentIndex()] = this.readBoolean();
+                    break;
+                case BYTE:
+                    values[serializableField.getRecordComponentIndex()] = this.readByte();
+                    break;
+                case CHAR:
+                    values[serializableField.getRecordComponentIndex()] = this.readChar();
+                    break;
+                case DOUBLE:
+                    values[serializableField.getRecordComponentIndex()] = this.readDouble();
+                    break;
+                case FLOAT:
+                    values[serializableField.getRecordComponentIndex()] = this.readFloat();
+                    break;
+                case INT:
+                    values[serializableField.getRecordComponentIndex()] = this.readInt();
+                    break;
+                case LONG:
+                    values[serializableField.getRecordComponentIndex()] = this.readLong();
+                    break;
+                case SHORT:
+                    values[serializableField.getRecordComponentIndex()] = this.readShort();
+                    break;
+            }
+        }
```

Comment:
Same as above. Why here are two enhanced for loops? Those could be merged into one or am I missing something?